### PR TITLE
Update AnnData file extensions (SCP-4557)

### PR DESCRIPTION
**Pull request by jlchang**
Anndata files (typically .h5ad) are also valid HDF5 files (.h5 and .hdf5). SCP has a few "other" files with .h5 suffix. While we expect most Anndata files to have the .h5ad suffix, ingest should try and see if Scanpy will successfully open files ending in .h5 and .hdf5.

This PR also updates the "Unsupported file format" error message that currently conveys expected file MIME type information. Instead we'll now return the expected file extensions which may be more actionable for our users.

Manual test:

activate the scp-ingest-pipeline repo virtualenv
then from the ingest directory of the scp-ingest-pipeline repo, perform this setup:

```
source ../scripts/setup_mongo_dev.sh <path/to/your/Github/token>
unset BARD_HOST_URL
```

Run the following ingest job to confirm that .h5 files ingest without error:
`python ingest_pipeline.py --study-id addedfeed000000000000000 --study-file-id dec0dedfeed1111111111111 --study-id 5d276a50421aa9117c982845 --study-file-id 5dd5ae25421aa910a723a337 ingest_h5ad --ingest-h5ad --h5ad-file ../tests/data/h5ad/test.h5`

You'll see information about the Anndata object and the job status is 'success'

> AnnData object with n_obs × n_vars = 238 × 187 backed at '../tests/data/h5ad/test.h5'
>     obs: 'submission_id', 'biosample_id', 'biosample_type', 'library_preparation_protocol', 'library_preparation_protocol__ontology_label', 'donor_id', 'species', 'species__ontology_label', 'mouse_strain', 'mouse_strain__ontology_label', 'organ', 'organ__ontology_label', 'brain_region', 'brain_region__ontology_label', 'sex', 'organism_age', 'organism_age__unit', 'organism_age__unit__ontology_label', 'disease', 'disease__ontology_label', 'seurat_clusters', 'cell_type', 'cell_type__ontology_label'
> distinct_id: 2f30ec50-a04d-4d43-8fd1-b136a2045079
> studyAccession: SCPdev
> fileName: 5dd5ae25421aa910a723a337
> fileType: input_validation_bypassed
> fileSize: 1
> trigger: dev-mode
> logger: ingest-pipeline
> appId: single-cell-portal
> status: success

Run the following ingest job to confirm that .h5 files ingest without error:
`python ingest_pipeline.py --study-id addedfeed000000000000000 --study-file-id dec0dedfeed1111111111111 --study-id 5d276a50421aa9117c982845 --study-file-id 5dd5ae25421aa910a723a337 ingest_h5ad --ingest-h5ad --h5ad-file ../tests/data/h5ad/bad.foo`

Confirm the file is rejected with an error message like:

> File type not detected for ../tests/data/h5ad/bad.foo, expected file endings are: .h5ad .h5 .hdf5

This PR supports SCP-4525 (frontend UI for h5ad ingest) and fulfills SCP-4557.

### CODE REVIEW

File: ingest/ingest_files.py [hunk 1]
```diff
@@ -82,10 +82,20 @@ class IngestFiles:
...
+        "text/plain": [".txt"],
+        "text/tab-separated-values": [".tsv"],
+        "dataframe": [".tsv"],
+        "application/x-hdf5": [".h5ad", ".h5", ".hdf5"],
+    }
+
     def __init__(self, file_path, allowed_file_types):
         self.file_path = file_path
-        # define filetype for h5ad file extension
+        # valid suffixes for Anndata (h5ad file extensions)
```
Comment (eweitz) [inline ingest/ingest_files.py:95]: Blocker: this comment seems to consider the file extensions `.h5` and `.hdf5` as an `.h5ad` file extension, which seems incorrect and confusing for maintainers.

Polish: Upstream folks refer to this as "AnnData" or "anndata"; rarely or never "Anndata" in my readings.

```suggestion
        # valid suffixes for AnnData
```

**Review by eweitz — APPROVED** [review]
Nice extension, and I like the user-friendlier error messaging.

I note one blocker, a semantic issue which seems likely to confuse maintainers, and thus important.  It seems trivial to fix and fairly straightforward (though I might be wrong), so I contingently approve.

Along those lines, but out-of-scope here, given our agreement on Friday that it's better to frame this new file type as "AnnData" instead of "H5ad" ([full rationale](https://github.com/broadinstitute/single_cell_portal_core/pull/1575#pullrequestreview-1055566982)), it seems worth renaming e.g. `test_h5ad.py`, `data/h5ad`, etc. to `test_anndata.py`, `data/anndata`, etc. some time soon.

**Review by bistline — APPROVED** [review]

File: ingest/monitor.py [hunk 2]
```diff
@@ -44,9 +44,25 @@ def setup_logger(logger_name, log_file, level=logging.DEBUG, format="default"):
...
+def bypass_mongo_writes():
+    """Check if developer has set environment variable to bypass writing data to MongoDB
+        BYPASS_MONGO_WRITES='yes'
+    """
+    if os.environ.get("BYPASS_MONGO_WRITES") is not None:
+        skip = os.environ["BYPASS_MONGO_WRITES"]
+        if skip == "yes":
+            return True
+        else:
+            return False
```
[the remark this answers is not in the document]
Comment (jlchang) [reply ingest/monitor.py]: `fa76fda`

**Comment by jlchang** [comment]
> 
> it's better to frame this new file type as "AnnData" instead of "H5ad" ([full rationale](https://github.com/broadinstitute/single_cell_portal_core/pull/1575#pullrequestreview-1055566982)), it seems worth renaming e.g. `test_h5ad.py`, `data/h5ad`, etc. to `test_anndata.py`, `data/anndata`, etc. some time soon.

Agreed, switched to "AnnData" instead of "H5ad" in `fa76fda`

File: ingest/ingest_files.py (hunk 1 above)
Comment (jlchang) [reply ingest/ingest_files.py]: Updated comment to be more accurate `fa76fda`

**Pull request merged**